lib/processors/manifestEnricher.js

```diff
@@ -0,0 +1,313 @@
+import semver from "semver";
+const {SemVer: Version} = semver;
+import {promisify} from "node:util";
+import path from "node:path/posix";
+import {getLogger} from "@ui5/logger";
+const log = getLogger("builder:processors:manifestEnricher");
+
+const APP_DESCRIPTOR_V22 = new Version("1.21.0");
+
+/**
+ * Transforms i18n_en_US.properties to en_US
+ *
+ * @param {string} bundleName
+ * @param {string} defaultBundleName
+ */
+function getLocale(bundleName, defaultBundleName) {
+	log.verbose("Resource bundle detected on " + bundleName);
+	bundleName = path.basename(bundleName, ".properties");
+	defaultBundleName = path.basename(defaultBundleName, ".properties");
+	if (bundleName === defaultBundleName) {
+		return "";
+	} else {
+		return bundleName.replace(defaultBundleName + "_", "");
+	}
+}
+
+/**
+ * Determines if manifest contain handlebars placeholder templates
+ *
+ * Only in case of placeholders in manifest.json the bundle configured
+ * in sap.app/i18n is requested by sap/ui/core/Manifest.js
+ *
+ * @param {object} manifest The manifest
+ * @returns {boolean}
+ */
+function hasManifestTemplates(manifest) {
+	const rManifestTemplate = /\{\{([^}}]+)\}\}/g;
+	return JSON.stringify(manifest).match(rManifestTemplate);
+}
+
+/**
+ * Transforms "ui5://my/fancy/app/i18n/i18n.properties" to "my.fancy.app.i18n.i18n"
+ *
+ * @param {string} bundleUrl A bundle Url
+ * @returns {string}
+ */
+function getBundleNameByUrl(bundleUrl) {
+	return bundleUrl.split("ui5://")[1].replaceAll("/", ".").split(".properties")[0];
```

Comment:
Is it possible that `ui5://` could be missing?
Maybe a `.replace("ui5://", "")` would be safer

Reply:
changed to `replaceAll` also last split in this line